core/drand_beacon.go

```diff
@@ -273,7 +273,7 @@ func (bp *BeaconProcess) transition(oldGroup *key.Group, oldPresent, newPresent
 	// NOTE: this limits the round time of drand - for now it is not a use
 	// case to go that fast
 	ctx := context.Background()
-	timeToStop := bp.group.TransitionTime - 1
+	timeToStop := bp.group.TransitionTime + int64(bp.group.Period.Seconds()) - 1
```

Comment:
Maybe add a comment saying
```suggestion
	// we stop the node at or after transition to make sure they are broadcasting 
	// their last partial before transition.
	timeToStop := bp.group.TransitionTime + int64(bp.group.Period.Seconds()) - 1
```